internal/s3select/select.go

```diff
@@ -546,6 +546,7 @@ func (s3Select *S3Select) Evaluate(w http.ResponseWriter) {
 			return false
 		}
 		outputQueue = outputQueue[:0]
+		bufPool.Put(buf)
```

Comment:
Ideally this should be just a `defer bufPool.Put(buf)` right after Get()

Reply:
Also, this is wrong since the buffer is sent to the writer and cannot be re-used.